### PR TITLE
Fix setting assignedSections to null when status is AVAILABLE

### PR DESCRIPTION
**Pull request by nmadruga**
[Front-end PR](https://github.com/openmsupply/application-manager-web-app/pull/1124) in progress

### CODE REVIEW

**Review by CarlosNZ — COMMENTED** [review]
I think we should probably leave it as non-nullable, default `[]`. The only reason we were changing it is so we could mutate it from the front-end, but we're using the trigger instead so no need. If it's NULL we don't need the trigger, if we're using the trigger its easier to deal with as [] rather than null.

Non-nullables are so much less hassle to deal with in the front-end, don't you think?

**Comment by CarlosNZ** [comment]
Also -- need to add migration code (can be part of 0.2.0 script, since we haven't released that yet)

**Comment by nmadruga** [comment]
Yes, Carl. Thanks for inputs. I agree. Will be changing this after stand-up meeting : )

File: database/buildSchema/28_review_assignment.sql [hunk 1]
```diff
@@ -68,12 +68,12 @@ CREATE TRIGGER review_assignment_trigger
     WHEN (NEW.trigger IS NOT NULL AND NEW.trigger <> 'PROCESSING' AND NEW.trigger <> 'ERROR')
     EXECUTE FUNCTION public.add_event_to_trigger_queue ();
 
--- TRIGGER (Listener) on review_assignment table: To set status accordingly to assignedSections
+-- TRIGGER (Listener) on review_assignment table: Set assignedSections to Null when changing status to AVAILABLE
```
Comment (CarlosNZ) [inline database/buildSchema/28_review_assignment.sql:71]:
```suggestion
-- TRIGGER (Listener) on review_assignment table: Set assignedSections to [] when changing status to AVAILABLE
```

File: database/buildSchema/28_review_assignment.sql [hunk 2]
```diff
@@ -68,12 +68,12 @@ CREATE TRIGGER review_assignment_trigger
...
 CREATE TRIGGER review_assignment_trigger2
     AFTER UPDATE OF trigger ON public.review_assignment
```
Comment (CarlosNZ) [inline database/buildSchema/28_review_assignment.sql:73]:
```suggestion
    AFTER UPDATE OF status ON public.review_assignment
```

I thought we already fixed this? 🤔

**Review by CarlosNZ — CHANGES_REQUESTED** [review]
Not sure how that incorrect field ended up in develop. Anyway, the trigger works once that's fixed.

**Review by CarlosNZ — APPROVED** [review]
Great thanks

**Pull request merged**